src/components/database.rs

```diff
@@ -57,11 +63,68 @@ impl DatabaseComponent {
         }
     }
 
-    pub fn get_repos(&self) -> &Option<DBRepositories> {
+    pub fn get_connection(db_connection: &Arc<Option<DBConnection>>) -> &DBConnection {
+        db_connection.as_ref().as_ref().unwrap()
+    }
+
+    pub async fn execute_query<'a>(
```

Comment:
I think we've talked about it but why not split this function into two functions? One for executing an ordinary query and one for executing a DB transaction. You are always passing the `Pool` when even if it's not needed.

I think we can have an `enum` type and replace the `pool` and `transaction` param with a `executor` param. We can do something like this: 

```rust 

pub enum Executor {
  Transaction(Transaction),
  Pool(Pool<Postgres>)
}

//........

pub async execute_query(query: Query<..>, executor: Executor) -> Result<> {
//.....
}


```

Reply:
I like the executor enum solution, will try it